src/main/java/org/assertj/core/error/ShouldContainExactly.java

```diff
@@ -92,36 +105,49 @@ public static ErrorMessageFactory shouldContainExactly(Object actual, Object exp
 
   public static ErrorMessageFactory shouldHaveSameSize(Object actual, Object expected, int actualSize,
                                                        int expectedSize, ComparisonStrategy comparisonStrategy) {
-    return StandardComparisonStrategy.instance().equals(comparisonStrategy) ?
-        new ShouldContainExactly(actual, expected, actualSize, expectedSize) :
-        new ShouldContainExactly(actual, expected, actualSize, expectedSize, comparisonStrategy);
+
+    Object[] actualArray = arrayOrIterableToArray(actual);
+    Object[] expectedArray = arrayOrIterableToArray(expected);
+    try {
+      List<Delta<String>> diffs = DIFF.diff(actualArray, expectedArray);
```

Comment:
This can throw (because internally it uses a BufferedReader), but I don't think it ever will.

Reply:
I would prefer we catch the `IOException` and rethrow it as a `RuntimeIOException` in `Diff` so that we don't pollute the calling code that can't do anything with the `IOException` anyway.